app/src/colorpalettewidget.cpp

```diff
@@ -558,7 +571,25 @@ void ColorPaletteWidget::updateItemColor(int itemIndex, QColor newColor)
     swatchPainter.drawTiledPixmap(0, 0, mIconSize.width(), mIconSize.height(), QPixmap(":/background/checkerboard.png"));
     swatchPainter.fillRect(0, 0, mIconSize.width(), mIconSize.height(), newColor);
 
-    ui->colorListWidget->item(itemIndex)->setIcon(colourSwatch);
+    QPen borderShadow(QColor(0, 0, 0, 200), 1, Qt::DotLine, Qt::FlatCap, Qt::MiterJoin);
+    QVector<qreal> dashPattern;
+    dashPattern << 4 << 4;
+    borderShadow.setDashPattern(dashPattern);
+    QPen borderHighlight(borderShadow);
+    borderHighlight.setColor(QColor(255, 255, 255, 200));
+    borderHighlight.setDashOffset(4);
+
+    QIcon swatchIcon;
+    swatchIcon.addPixmap(colourSwatch, QIcon::Normal);
+
+    // Draw selection border
+    swatchPainter.setPen(borderHighlight);
+    swatchPainter.drawRect(0, 0, mIconSize.width() - 1, mIconSize.height() - 1);
+    swatchPainter.setPen(borderShadow);
+    swatchPainter.drawRect(0, 0, mIconSize.width() - 1, mIconSize.height() - 1);
+    swatchIcon.addPixmap(colourSwatch, QIcon::Selected);
```

Comment:
Minor thing: selection border should imo. only be shown in grid mode. There's already normal cell selection highlight in list mode.

Reply:
I agree that it isn't necessary in list mode, but I think it's still useful because the selection highlight differs based on the operating system. Here's a screenshot of the QListWidget from the qt docs:

![](https://doc.qt.io/qt-5/images/windows-listview.png)

Its style has much less emphasis on a selection than on a mac and a selection border may look better in styles like this.

Also I feel like it would be a bit of a pain to change because you would have to change the icons each time you toggle the mode, and either regenerate or store and retrieve the selection pixmap for each color when switching to grid mode. So part of the reason why I didn't do this is just laziness 😉

Reply:
The list is updated on every click anyway so the required changes are minimal to hide the border. The style also seems pretty consistent across systems too, so aside from changing the Qt Theme from source code, the style should be somewhat consistent. 

The image you provided is not what it looks like in Pencil2D though, regardless the operation system as far as I know, this is how it looks through wine:
![image](https://user-images.githubusercontent.com/1045397/43312207-db6b61ae-918c-11e8-84c7-18afb238156a.png)

The only changes that are required is adding if statements around your selection border painting. ~290 and ~650

Reply:
For some reason I was thinking that refreshColorList wasn't called on mode change, but I can see now that it is, and you're right it's very simple to change. I have pushed the changes needed to show the selection border in grid mode only.